rosbag2/include/rosbag2/reader.hpp

```diff
@@ -43,22 +43,17 @@ namespace rosbag2
 class BaseReaderInterface;
 
 /**
- * The SequentialReader allows opening and reading messages of a bag. Messages will be read
- * sequentially according to timestamp.
+ * The Reader allows opening and reading messages of a bag.
  */
-class ROSBAG2_PUBLIC SequentialReader
+class ROSBAG2_PUBLIC Reader final
 {
 public:
-  explicit SequentialReader(
-    std::unique_ptr<rosbag2_storage::StorageFactoryInterface> storage_factory =
-    std::make_unique<rosbag2_storage::StorageFactory>(),
-    std::shared_ptr<SerializationFormatConverterFactoryInterface> converter_factory =
-    std::make_shared<SerializationFormatConverterFactory>());
+  Reader(std::unique_ptr<BaseReaderInterface> reader_impl);
```

Comment:
```suggestion
  explicit Reader(std::unique_ptr<BaseReaderInterface> reader_impl);
```

Reply:
addressed in https://github.com/aws-ros-dev/rosbag2/pull/4/commits/5a670023464e57f2c12a2379b0bac2758cdf7e1d